plugins/MantisGraph/MantisGraph.php

```diff
@@ -140,13 +154,20 @@ function graph_filter_menu() {
 	function resources() {
 		if( current( explode( '/', gpc_get_string( 'page', '' ) ) ) === $this->basename ) {
 			if( config_get_global( 'cdn_enabled' ) == ON ) {
-				html_javascript_cdn_link('https://cdnjs.cloudflare.com/ajax/libs/Chart.js/' . CHARTJS_VERSION . '/Chart.min.js', CHARTJS_HASH);
-				html_javascript_cdn_link('https://cdnjs.cloudflare.com/ajax/libs/Chart.js/' . CHARTJS_VERSION . '/Chart.bundle.min.js', CHARTJSBUNDLE_HASH);
-				html_javascript_cdn_link('https://cdn.jsdelivr.net/npm/chartjs-plugin-colorschemes@' . self::CHARTJS_COLORSCHEMES_VERSION . '/dist/chartjs-plugin-colorschemes.min.js', self::CHARTJS_COLORSCHEMES_HASH );
+				$t_cdn = 'https://cdn.jsdelivr.net/npm/%s@%s/dist/';
+
+				# Chart.js library
+				$t_link = sprintf( $t_cdn, self::CHARTJS_LIB, self::CHARTJS_VERSION );
+				html_javascript_cdn_link( $t_link . self::CHARTJS_FILE, self::CHARTJS_HASH );
+				html_javascript_cdn_link( $t_link . self::CHARTJSBUNDLE_FILE, self::CHARTJSBUNDLE_HASH );
+
+				# Chart.js color schemes plugin
+				$t_link = sprintf( $t_cdn, self::CHARTJS_COLORSCHEMES_LIB, self::CHARTJS_COLORSCHEMES_VERSION );
+				html_javascript_cdn_link( $t_link . self::CHARTJS_COLORSCHEMES_FILE, self::CHARTJS_COLORSCHEMES_HASH );
 			} else {
-				echo '<script type="text/javascript" src="' . plugin_file('Chart-' . CHARTJS_VERSION . '.min.js') . '"></script>';
-				echo '<script type="text/javascript" src="' . plugin_file('Chart.bundle-' . CHARTJS_VERSION . '.min.js') . '"></script>';
-				echo '<script type="text/javascript" src="' . plugin_file( 'chartjs-plugin-colorschemes.min.js' ) . '"></script>';
+				echo '<script type="text/javascript" src="' . plugin_file( self::CHARTJS_FILE ) . '"></script>';
```

Comment:
Isn't it better to go on using the versions also for non-CDS to prevent caching side effects after upgrade?

Reply:
I did not think of that. Good point, will change.